IronSoftware.Drawing/IronSoftware.Drawing.Common/AnyBitmap.cs

```diff
@@ -485,13 +492,24 @@ public static AnyBitmap FromStream(Stream stream)
         {
             return new AnyBitmap(stream);
         }
+
+        /// <summary>
+        /// Construct a new Bitmap from binary data (byte span).
+        /// </summary>
+        /// <param name="span">A byte span of image data in any common format.</param>
+        /// <seealso cref="AnyBitmap"/>
+        public AnyBitmap(in ReadOnlySpan<byte> span)
+        {
+            LoadImage(span);
+        }
+
         /// <summary>
         /// Construct a new Bitmap from binary data (bytes).
         /// </summary>
         /// <param name="bytes">A ByteArray of image data in any common format.</param>
         /// <seealso cref="FromBytes"/>
         /// <seealso cref="AnyBitmap"/>
-        public AnyBitmap(byte[] bytes)
+        public AnyBitmap(in byte[] bytes)
```

Comment:
Interesting thing, but the Microsoft doc states, that this creates a temporary variable with a copy of the parameter, and warns that if large in size structs are passed that way - it can actually impact perf negatively. Which might apply to images, since those might sometimes get very large.

Just wondering if this could be of concern here.

[Doc in question](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/method-parameters#in-parameter-modifier)

Reply:
Second what Artem mentions. `in` would likely make a copy to signify that it's input only argument. I think `byte[]` is already fine in this case.

Even if we pass it as `Span<T>` which might be more expressive, but `byte[]` already has size property that we can get, and it always passed as reference for array.

Reply:
nice, I saw "passes as ref" and didn't read the rest of the docs. nice catch.

the `in` keywords have been removed